src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeHandlerTest.java

```diff
@@ -144,7 +145,47 @@ void should_handle_and_save_envelope() {
         assertThat(envelope.getValue().getPayments()).isEqualTo(inputEnvelope.payments);
         assertThat(envelope.getValue().getNonScannableItems()).isEqualTo(inputEnvelope.nonScannableItems);
         assertThat(envelope.getValue().getContainer()).isEqualTo(CONTAINER_NAME);
-        assertThat(envelope.getValue().getRescanFor()).isEqualTo(RESCAN_FOR_FILE_NAME);
+
+        verifyNoInteractions(fileRejector);
+        verifyNoMoreInteractions(envelopeProcessor);
+    }
+
+    @Test
+    void should_retrow_exception_from_ocr_validator() {
+        // given
+        given(ocrValidator.assertOcrDataIsValid(inputEnvelope, blobClient, LEASE_ID))
+            .willThrow(new RuntimeException("msg"));
+        given(containerMappings.getMappings()).willReturn(mappings);
+
+        // when
+        RuntimeException exception = catchThrowableOfType(
+            () -> envelopeHandler.handleEnvelope(
+                CONTAINER_NAME,
+                FILE_NAME,
+                pdfs,
+                inputEnvelope,
+                blobClient,
+                LEASE_ID
+            ),
+            RuntimeException.class
+        );
+
+        // then
+        verify(envelopeValidator).assertZipFilenameMatchesWithMetadata(inputEnvelope, FILE_NAME);
+        verify(envelopeValidator).assertContainerMatchesJurisdictionAndPoBox(
+            containerMappings.getMappings(), inputEnvelope, CONTAINER_NAME
+        );
+        verify(envelopeValidator).assertServiceEnabled(inputEnvelope, containerMappings.getMappings());
+        verify(envelopeValidator).assertEnvelopeContainsOcrDataIfRequired(inputEnvelope);
+        verify(envelopeValidator).assertEnvelopeHasPdfs(inputEnvelope, pdfs);
+        verify(envelopeValidator).assertDocumentControlNumbersAreUnique(inputEnvelope);
+        verify(envelopeValidator).assertPaymentsEnabledForContainerIfPaymentsArePresent(
+            inputEnvelope, paymentsEnabled, containerMappings.getMappings()
+        );
+        verify(envelopeValidator).assertEnvelopeContainsDocsOfAllowedTypesOnly(inputEnvelope);
+        verify(envelopeProcessor).assertDidNotFailToUploadBefore(inputEnvelope.zipFileName, CONTAINER_NAME);
+
+        assertThat(exception.getMessage()).isEqualTo("msg");
```

Comment:
Are those verification required? This test is checking if the EnvelopeHandler rethrows the exceptions, calling those other validator is side effect not related to the tests and checked already in the test above. I would probably skip those.